doc/ebpf/env.rst

```diff
@@ -0,0 +1,630 @@
+Building Linux Image for Working with eBPF in QEMU
+==================================================
+
+.. contents:: :local:
+
+Overview
+--------
+
+This is a comprehensive guide about creating an environment for 
+building, running and debugging eBPF programs for ARC processors using
+`GNU toolchain <https://github.com/foss-for-synopsys-dwc-arc-processors/arc-gnu-toolchain>`_
+and `QEMU <https://github.com/foss-for-synopsys-dwc-arc-processors/qemu>`_. Though we consider
+ARC HS 3x/4x on QEMU as a reference platform, the same guide is applicable for boards
+like HS Development Kit.
+
+This guide consists of these steps:
+
+1. Preparing your Linux host for building Buildroot image and Linux kernel and for building third-party tools and libraries.
```

Comment:
1. Preparing your Linux host for building rootfs (Buildroot), Linux kernel, third-party tools and libraries.